pageserver/src/layered_repository/delta_layer.rs

```diff
@@ -293,11 +293,16 @@ impl Layer for DeltaLayer {
     /// it will need to be loaded back.
     ///
     fn unload(&self) -> Result<()> {
-        let mut inner = self.inner.lock().unwrap();
-        inner.page_version_metas = VecMap::default();
-        inner.relsizes = VecMap::default();
-        inner.loaded = false;
-        Ok(())
+        match self.inner.try_write() {
+            Ok(mut inner) => {
+                inner.page_version_metas = VecMap::default();
+                inner.relsizes = VecMap::default();
+                inner.loaded = false;
+                Ok(())
+            }
+            Err(TryLockError::WouldBlock) => Ok(()),
```

Comment:
Shouldn't we try to signal that we did not unload the relation's in-memory datastructures, through e.g. `Result<bool>`?

Reply:
The caller doesn't care, so doesn't really matter.